### PR TITLE
Normalize paths in transformers to guard against bad paths from upstream transformers

### PR DESCRIPTION
**Pull request by justinfagnani**

### CODE REVIEW

**Comment by justinfagnani** [comment]
A test that would hit the actual case we're seeing here would have to test the whole build pipeline. I aspire to that...

**Comment by FredKSchott** [comment]
ugh, good catch. LGTM if tests pass

**Comment by justinfagnani** [comment]
Helps #178

**Comment by justinfagnani** [comment]
@FredKSchott I caught a few more places that needed normalization guards. build seems like it's working for the app template on Windows 10 now. PTAL

**Comment by FredKSchott** [comment]
code looks good!

I'm having trouble getting build to run on both windows and mac right now (`no resolver found for /bower_components/webcomponentsjs/webcomponents-lite.js`), trying to diagnose but it looks like its not a problem with this branch.

**Comment by FredKSchott** [comment]
On `polymer build` on windows for the shop template:

```

debug:   no file for src/src/shop-button.html :(
debug:   No file found for src/src/shop-button.html
debug:   no file for src/src/shop-image.html :(
debug:   No file found for src/src/shop-image.html
debug:   no file for src/src/shop-common-styles.html :(
debug:   No file found for src/src/shop-common-styles.html
debug:   no file for src/src/shop-select.html :(
debug:   No file found for src/src/shop-select.html
debug:   no file for src/src/shop-form-styles.html :(
debug:   No file found for src/src/shop-form-styles.html
... (plenty more)

error:   Promise rejection: Error: no resolver found for src/bower_components/polymer/polymer.html
error:   Error: no resolver found for src/bower_components/polymer/polymer.html
```

Not sure if this is related but worth sharing

File: test/build/html-project_test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+const assert = require('chai').assert;
+const path = require('path');
+const stream = require('stream');
+const File = require('vinyl');
+
+const HtmlProject = require('../../lib/build/html-project').HtmlProject;
+
+suite('HtmlProject', () => {
+
+  test.only('deals with bad paths', (done) => {
```
Comment (FredKSchott) [inline test/build/html-project_test.js]: remove `.only`

File: test/build/html-project_test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+          file.path = file.path.replace('\\', '/');
+        }
+      })
+      .pipe(project.rejoin)
+      .on('data', (file) => {
+        let contents = file.contents.toString();
+        assert.equal(contents, source);
+      })
+      .on('finish', () => done());
+
```
Comment (FredKSchott) [inline test/build/html-project_test.js]: Just confirming: do we need a catch handler on this stream, or does mocha automatically pick them up as thrown errors?

**Comment by FredKSchott** [comment]
test lgtm once that `.only` is removed

File: test/build/html-project_test.js (hunk 1 above)
Comment (justinfagnani) [inline test/build/html-project_test.js]: done

File: test/build/html-project_test.js (hunk 2 above)
Comment (justinfagnani) [inline test/build/html-project_test.js]: it timed out on failure, but I added a on/error handler

**Pull request merged**